src/dataEditor/dataEditorClient.ts

```diff
@@ -443,13 +460,27 @@ export class DataEditorClient implements vscode.Disposable {
             startOffset,
             length
           )
+          const characterCount = await countCharacters(
+            this.omegaSessionId,
+            startOffset,
+            length
+          )
           await this.panel.webview.postMessage({
             command: MessageCommand.profile,
             data: {
               startOffset: startOffset,
               length: length,
               byteProfile: byteProfile,
               numAscii: numAscii(byteProfile),
+              characterCount: {
```

Comment:
I feel like the `characterCount` field should be a single object given that there's already a `CharacterCountData` class defined in the DataMetrics.svelte file. It will just have to be moved to a file in which both the extension and UI can import from.